packages/emitter-framework/src/java/components/enum.tsx

```diff
@@ -0,0 +1,76 @@
+import { mapJoin } from "@alloy-js/core";
+import * as jv from "@alloy-js/java";
+import { Enum } from "@typespec/compiler";
+
+export interface EnumProps {
+  type: Enum;
+}
+
+export function EnumDeclaration({ type }: EnumProps) {
+  const members = Array.from(type.members.values());
+
+  let valueType: string | undefined;
+  let isConsistent = true;
+
+  members.forEach((member, index) => {
+    if (member.value !== undefined && member.value !== null) {
+      const currentType = typeof member.value === "string" ? "String" : "int";
+
+      if (valueType === undefined) {
+        valueType = currentType;
+      } else if (valueType !== currentType) {
+        isConsistent = false;
+      }
+    }
+  });
+
+  // TypeSpec allows you to mix and match values, for a Java enum have to have one value
+  if (!isConsistent) {
+    throw new Error("All enum values must have the same type");
```

Comment:
When we see an enum type with mixed value types, should we convert that into a class instead and have a list of static constants instead?

```java
public final class FooType {
    public static String STRING_VALUE_1 = "val1";
    public static String STRING_VALUE_2 = "val2";
    public static Integer INTEGER_VALUE_1 = 5;
    ...

   private FooType() { } // private constructor to not allow instantiation
}

Reply:
This is a good idea! I've added some code for it except using the Alloy <Variable /> component forces the name to be lowercase, so stringValue1, instead of STRING_VALUE_1, so will just had to add an issue for fixing this